client/db.go

```diff
@@ -314,6 +335,20 @@ type Tx struct {
 	txn *Txn
 }
 
+// SetSnapshotIsolation sets the transaction's isolation type to
+// snapshot. Transactions default to serializable isolation. The
+// isolation must be set before any operations are performed on the
+// transaction.
+//
+// TODO(pmattis): This isn't tested yet but will be as part of the
+// conversion of client_test.go.
+func (tx *Tx) SetSnapshotIsolation() {
+	// TODO(pmattis): Panic if the transaction has already had
+	// operations run on it. Needs to tie into the Txn reset in case of
+	// retries.
+	tx.txn.txn.Isolation = proto.SNAPSHOT
```

Comment:
Yeah. This will likely get simplified as the implementation improves. Bigger question is whether the API is good. Setting of the isolation level is rare so it seemed worthwhile to keep it out of the `DB.Tx` method arguments. Perhaps there should be a `DB.SnapshotTx`.